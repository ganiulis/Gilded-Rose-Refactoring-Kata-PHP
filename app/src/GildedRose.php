<?php

declare(strict_types=1);

namespace GildedRose;

use GildedRose\Updater;

/**
 * Used for updating an array of Items.
 */
final class GildedRose
{
    private const AGED_BRIE = 'Aged Brie';
    private const BACKSTAGE_PASSES = 'Backstage passes to a TAFKAL80ETC concert';
    private const CONJURED = 'Conjured Item';
    private const SULFURAS = 'Sulfuras, Hand of Ragnaros';

    /**
     * Updated quality of selected array of Items
     *
     * @param array $items Items array to be updated
     * @return array updated Items array
     */
    public function updateItems(array $items): array
    {
        foreach ($items as $item) {
            $itemType = preg_match('/\bConjured\b/i', $item->name) ? self::CONJURED : $item->name;

            switch($itemType) {
                case self::AGED_BRIE:
                    $this->itemUpdater = new Updater\BrieUpdater;
                    break;
                case self::BACKSTAGE_PASSES:
                    $this->itemUpdater = new Updater\BackstageUpdater;
                    break;
                case self::CONJURED:
                    $this->itemUpdater = new Updater\ConjuredUpdater;
                    break;
                case self::SULFURAS:
                    $this->itemUpdater = new Updater\SulfurasUpdater;
                    break;
                default:
                    $this->itemUpdater = new Updater\DefaultUpdater;
                    break;
                }
            $this->itemUpdater->updateItem($item);
        }
        return $items;
    }
}
