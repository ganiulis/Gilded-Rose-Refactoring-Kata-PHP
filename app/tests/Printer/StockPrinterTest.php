<?php

namespace Tests\Printer;

use ApprovalTests\Approvals;
use GildedRose\Item;
use GildedRose\Printer\StockPrinter;
use PHPUnit\Framework\TestCase;

class StockPrinterTest extends TestCase
{
    public function testStockPrinter(): void
    {
        $printer = new StockPrinter();

        $testItems = [
            new Item('foo', 4, 3),
            new Item('bar', 5, 2),
            new Item('zim', 6, 1),
            new Item('gir', 7, 0)
        ];
        
        ob_start();

        $printer->printIntro();

        $printer->printSummary($testItems, 3);
        
        $output = ob_get_contents();

        Approvals::verifyString($output);

        ob_end_clean();
    }
}
