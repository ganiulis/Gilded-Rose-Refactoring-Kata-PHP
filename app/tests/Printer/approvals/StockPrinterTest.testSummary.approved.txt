-------- day 4321 --------
name, sellIn, quality
alpha, 5, 0
bravo, 4, 1
charlie, 3, 2
delta, 2, 3
echo, 1, 4
foxtrot, 0, 5

